src/app/clusters/refrigerator-alarm-server/refrigerator-alarm-server.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::RefrigeratorAlarm;
+using namespace chip::app::Clusters::RefrigeratorAlarm::Attributes;
+using namespace chip::DeviceLayer;
+using chip::Protocols::InteractionModel::Status;
+
+static Status resetHandler(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath, const chip::BitMask<AlarmMap> alarms, const chip::Optional<chip::BitMask<AlarmMap>> mask);
+
+static Status resetHandler(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath, const chip::BitMask<AlarmMap> alarms, const chip::Optional<chip::BitMask<AlarmMap>> mask)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    EmberAfStatus status;
+    chip::BitMask<AlarmMap> state = 0;
+    status = State::Get(endpoint, &state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+    
+    uint32_t alarmValue = alarms.Raw();
+    uint32_t stateValue = state.Raw();
+
+    // Flip the bits of alarms (i.e. ~alarms)
+    alarmValue = 0xFFFF ^ alarmValue;
+
+    // Reset state from active to inactive
+    stateValue = stateValue & alarmValue;
+
+    state.SetRaw(stateValue);
```

Comment:
This seems like a somewhat complicated way to write:
```suggestion
    state.Clear(alarms);
```
Unless I'm just missing something?